modules/hitlnext/src/backend/repository.ts

```diff
@@ -464,4 +472,34 @@ export default class Repository {
       { count: conditions.limit, sortOrder: [{ column: 'id', desc: true }] }
     )
   }
+
+  deleteHandoff = async (id: string) => {
+    return this.bp.database.transaction(async trx => {
+      const handoff = await this.getHandoff(id)
+
+      // We want delete the events since it contains
+      // the user's conversation.
+      await trx(EVENT_TABLE_NAME)
+        .del()
+        .whereIn('threadId', [handoff.agentThreadId, handoff.userThreadId])
+
+      await trx(WEB_MESSAGE_TABLE_NAME)
+        .del()
+        .whereIn('conversationId', [handoff.agentThreadId, handoff.userThreadId])
+
+      await trx(WEB_CONVERSATION_TABLE_NAME)
+        .del()
+        .whereIn('id', [handoff.agentThreadId, handoff.userThreadId])
+
+      // We want to delete rows from this table since it
+      // contains user info like its timezone and language.
+      await trx(SRV_CHANNEL_USER_TABLE_NAME)
```

Comment:
really ? we delete the user itself ? don't we only want to delete a specific user conversation ?  keep in mind that a user can have multiple conversation in a channel

Reply:
Agreed.